### PR TITLE
Bug 1957498: Clean up tech debt

### PR DESCRIPTION
**Pull request by hexfusion**
This PR removes a few areas of code that performed various checks and cleanups for 4.7-> 4.8 and are no longer needed in 4.9.

### CODE REVIEW

**Comment by hexfusion** [comment]
cc @marun

**Comment by hexfusion** [comment]
/hold
Still going to need to delete the v1beta object need to think about this...

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -350,3 +324,61 @@ var CertSecrets = []installer.UnrevisionedResource{
...
+		if err != nil {
+			klog.Warningf("Error retrieving legacy pod disruption budget: %v", err)
+			return
+		}
+		if legacyPDB.ObjectMeta.DeletionTimestamp != nil {
+			// Done - resource has been marked for deletion
+			return
+		}
+
+		// Check that the deployment managing the apiserver pods has the correct number of replicas
```
Comment (marun) [inline pkg/operator/starter.go:365]: nit: s/apiserver/etcd/

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -350,3 +324,61 @@ var CertSecrets = []installer.UnrevisionedResource{
...
+		etcdOperatorQuorumGuard, err := etcdOClient.Get(ctx, resourceName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			// No available replicas if the deployment doesn't exist
+			return
+		}
+		if etcdOperatorQuorumGuard.Status.AvailableReplicas == etcdOperatorQuorumGuard.Status.Replicas {
```
Comment (marun) [inline pkg/operator/starter.go:371]: Maybe comment or make it clear in the code why `AvailableReplicas == Replicas` is indicative of not having all available replicas?

**Review by marun — COMMENTED** [review]
Yay tackling tech debt!  A couple of nits inline. Also, bz?

File: pkg/operator/starter.go (hunk 2 above)
Comment (hexfusion) [reply pkg/operator/starter.go]: thanks clearly wrong operator 0:)

**Comment by marun** [comment]
/lgtm

Most cleanup code I've seen tends to run periodically (e.g. the secret removal code that this PR attempts to remove every 30 minutes). Is it not desirable to do similarly for pdb?

**Comment by hexfusion** [comment]
> /lgtm
> 
> Most cleanup code I've seen tends to run periodically (e.g. the secret removal code that this PR attempts to remove every 30 minutes). Is it not desirable to do similarly for pdb?

We check every minute https://github.com/openshift/cluster-etcd-operator/pull/631/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R383 I might be misunderstanding.

**Comment by marun** [comment]
> We check every minute https://github.com/openshift/cluster-etcd-operator/pull/631/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6R383 I might be misunderstanding.

Aha, it's me doing the misunderstanding!

**Comment by hexfusion** [comment]
/hold upgrade has some broken assumptions

```
 event happened 41 times, something is wrong: ns/openshift-etcd-operator deployment/etcd-operator - reason/LegacyPDBRemoved legacy v1beta1 pod disruption budget resource removed
event happened 41 times, something is wrong: ns/openshift-etcd-operator deployment/etcd-operator - reason/ModifiedQuorumGuardPDB etcd-quorum-guard pdb was modified
```

**Comment by hexfusion** [comment]
infra 
/test e2e-agnostic

**Comment by lilic** [comment]
It still has hold, hence the no retesting.

/lgtm

/retest

**Comment by hexfusion** [comment]
Minor upgrade test https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-gcp/1423749036162158592

**Comment by hexfusion** [comment]
Minor upgrade test https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-gcp/1423749036162158592

**Comment by ravisantoshgudimetla** [comment]
/retitle Bug 1957498: Clean up tech debt

**Comment by ravisantoshgudimetla** [comment]
/retitle "Bug 1957498: Clean up tech debt"

**Review by ravisantoshgudimetla — APPROVED** [review]
/lgtm

Thank you for the patch @hexfusion

**Pull request merged**